### DISCUSSION TITLE
Ruleset enforcing status checks can be bypassed by anyone

### DISCUSSION THREAD

**Discussion by serpro69** [community/community#121511] [Repositories] [answered]
### Select Topic Area

Bug

### Body

We have a ruleset on the main branch that requires status checks to pass before merge.

![image](https://github.com/community/community/assets/22973227/588c51c0-fbb7-4f9d-b6c9-ff28e6a4f4af)

Those status checks didn't pass in the PR (they were skipped because an dependency status check failed) and we could merge the PR with the account that is _not_ in the "bypass list"

![image](https://github.com/community/community/assets/22973227/cb1f024a-d075-4991-a22b-4e9c2423d29f)

**Comment by collinmcneese** [comment, marked as answer]
Hello!  Just wanted to drop a note here that in the screenshots, it looks like the `check-pr/lint` and `check-pr/test` status checks were being checked and in the subsequent screenshot it looks like those jobs show as a status of `skipped` .

Skipped jobs count as `success` when it comes to status checks.  There are a number of reasons for this, one of which is so that jobs can be [conditional](https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution) with `if:` type checks at the job level and skipped jobs then status as `success` rather than `failure`.

From the [docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks):
>Note: A job that is skipped will report its status as "Success". It will not prevent a pull request from merging, even if it is a required check.
>

**Comment by serpro69** [comment]
Oh, I see. Thank you for the clarification!

**Comment by Smaug123** [comment]
Is there a way to obtain the intended behaviour - that is, a "required status check" which is actually required?
